xbmc/Application.cpp

```diff
@@ -2778,6 +2779,26 @@ bool CApplication::OnMessage(CGUIMessage& message)
     }
     break;
 
+    case GUI_MSG_PLAY_TRAILER:
+    {
+      const CFileItemPtr item = std::static_pointer_cast<CFileItem>(message.GetItem());
+      std::unique_ptr<CFileItem> trailerItem =
+          ContentUtils::GeneratePlayableTrailerItem(item, g_localizeStrings.Get(20410));
+
+      if (item->IsPlayList())
+      {
+        CFileItemList* l = new CFileItemList; //don't delete,
+        l->Add(std::move(trailerItem));
+        CServiceBroker::GetAppMessenger()->PostMsg(TMSG_MEDIA_PLAY, -1, -1, static_cast<void*>(l));
```

Comment:
I think we should redo this function to have no raw pointers. Avoid `new` using `std::make_unique`, minimize allocations (we can save one here), and then when passing the pointer off, call `release()`. We won't need a comment that way, as ownership will be explicit.